### PR TITLE
feat: create min dust fee setting

### LINKED ISSUE #4501: Add setting to ignore dust
The wallet needs a setting to ignore dust UTXOs

### LINKED ISSUE #5807: Default output selection criteria can be abused to increase the sender's fee
The output selection mechanism can be abused to increase the sender's
transaction fee in scenarios where the adversary can control the amount to be
transferred.
The default output selection mechanism can use two modes depending on the
amount to be transferred. If the amount is larger than the maximum output value,
the order will be descendant. Otherwise, it will be ascendant, meaning it will
select smaller outputs first.

Therefore, an adversary able to control the amount to be transferred can force the
sender to use dust values to increase the transaction fee. The dust input values to
be used by the sender could be previously sent by the adversary.
Even though the wallet admits configuring a flag to drop the transaction in case
the fees exceed the amount to be transferred, it can still be abused to spend an
amount of fees slightly lower than the value to be transferred.
Recommendation
Ignore dust amounts in the output selection mechanism.
Also, consider implementing algorithms such as Branch and Bound which
minimize the creation of change outputs which can be of dust amounts, by trying to
find the exact match for transaction amounts in the UTXO set.

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Create a setting to ignore dust below a certain value/ 

Motivation and Context
---

See https://github.com/tari-project/tari/issues/5807

Fixes: https://github.com/tari-project/tari/issues/5807
Fixes: #4501

### CODE REVIEW

File: base_layer/wallet/src/output_manager_service/input_selection.rs [hunk 1]
```diff
@@ -64,6 +67,7 @@ impl UtxoSelectionCriteria {
         Self {
             filter: UtxoSelectionFilter::SpecificOutputs { commitments },
             ordering: UtxoSelectionOrdering::Default,
+            min_dust: 0,
```
Comment (brianp) [inline base_layer/wallet/src/output_manager_service/input_selection.rs:70]: Should `specific` have other unknown qualifiers than those specified?

File: common/config/presets/d_console_wallet.toml [hunk 2]
```diff
@@ -133,6 +133,8 @@ transaction_event_channel_size = 25000
 # transaction amount. Set this value to `false` to allow spending of "dust" UTXOs for small valued transactions
 # (default = true).
 prevent_fee_gt_amount = false
+# Ignores dust below this value, value in micro MinoTari
```
Comment (brianp) [inline common/config/presets/d_console_wallet.toml]:
```suggestion
# Ignores dust below this value, value in micro MinoTari, defaults to 100
```

Just to remove ambiguity around the fact this is happening whether this option is enabled or not.

**Review by brianp — COMMENTED** [review]
Ack

Mostly looks good. Two comments.

File: base_layer/wallet/src/output_manager_service/input_selection.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/wallet/src/output_manager_service/input_selection.rs:70]: No just that, you are selecting very specific commitments, so there is no need for a dust filter

Comment (brianp) [reply base_layer/wallet/src/output_manager_service/input_selection.rs:70]: 🙃 

I clearly glossed over the actual value.

**Pull request merged**